### PR TITLE
AggregatorV1 - leave only genesis instance with no limit

### PR DESCRIPTION
**Pull request by igor-aptos**
## Description
* only allowing AggregatorV1 to be created with limit == u64::MAX
* only allowing optional_aggregator to be created within genesis (for coin APT total supply). removing upgrade / switch 

## Type of Change
- [ ] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Other (specify)

## Checklist
- [ ] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I identified and added all stakeholders and component owners affected by this change as reviewers
- [ ] I tested both happy and unhappy path of the functionality
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: aptos-move/framework/aptos-framework/sources/aggregator/aggregator_factory.move [hunk 1]
```diff
@@ -16,6 +16,11 @@ module aptos_framework::aggregator_factory {
     /// Aggregator factory is not published yet.
     const EAGGREGATOR_FACTORY_NOT_FOUND: u64 = 1;
 
+    /// Aggregator V1 only supports limit == MAX_U128
```
Comment (georgemitenkov) [inline aptos-move/framework/aptos-framework/sources/aggregator/aggregator_factory.move:19]: nit `.` missing in the end

File: aptos-move/framework/aptos-framework/sources/aggregator/optional_aggregator.move [hunk 2]
```diff
@@ -16,6 +16,9 @@ module aptos_framework::optional_aggregator {
     /// Aggregator feature is not supported. Raised by native code.
     const EAGGREGATOR_UNDERFLOW: u64 = 2;
 
+    /// OptionalAggregator (Agg V1) switch not supported any more
```
Comment (georgemitenkov) [inline aptos-move/framework/aptos-framework/sources/aggregator/optional_aggregator.move:19]: nit: `.` is missing in the end.

File: aptos-move/framework/aptos-framework/sources/aggregator/optional_aggregator.move [hunk 3]
```diff
@@ -84,46 +87,8 @@ module aptos_framework::optional_aggregator {
...
-        optional_aggregator: &mut OptionalAggregator
-    ): u128 {
-        let integer = option::extract(&mut optional_aggregator.integer);
-        let limit = limit(&integer);
-        destroy_integer(integer);
-        let aggregator = aggregator_factory::create_aggregator_internal(limit);
-        option::fill(&mut optional_aggregator.aggregator, aggregator);
-        limit
+    public fun switch(_optional_aggregator: &mut OptionalAggregator) {
+        abort error::invalid_state(ESWITCH_DEPRECATED)
```
Comment (georgemitenkov) [inline aptos-move/framework/aptos-framework/sources/aggregator/optional_aggregator.move:91]: I am wondering if we can unify error codes (e.g., define them in `features.move`) for all deprecated features like this, so we do not need to define new constants, and just call `features::deprecated()` which will abort, or something like this?

**Review by georgemitenkov — APPROVED** [review]

File: aptos-move/framework/aptos-framework/sources/aggregator/optional_aggregator.move (hunk 3 above)
Comment (igor-aptos) [reply aptos-move/framework/aptos-framework/sources/aggregator/optional_aggregator.move:91]: you cannot link to a constant from another file. though we could have a function abort_deprecated_feature() or something like that, if that's what you meant? but that would also be hard to debug - which deprecated feature is reporting the error?

Comment (georgemitenkov) [reply aptos-move/framework/aptos-framework/sources/aggregator/optional_aggregator.move:91]: Yes, I meant the function, local constant is also fine, was proposing a function to unify things but extra constant is ok

File: aptos-move/framework/aptos-framework/sources/aggregator/optional_aggregator.move [hunk 4]
```diff
@@ -192,7 +157,7 @@ module aptos_framework::optional_aggregator {
     }
```
Comment (gelash) [inline aptos-move/framework/aptos-framework/sources/aggregator/optional_aggregator.move:157]: supernit: can we add a unit test for switch aborting?

**Review by gelash — APPROVED** [review]

**Pull request merged**